instrument-starters/opentracing-spring-cloud-redis-starter/src/test/java/io/opentracing/contrib/spring/cloud/redis/IntegrationTest.java

```diff
@@ -0,0 +1,124 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.spring.cloud.redis;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import ai.grakn.redismock.RedisServer;
+import io.opentracing.Scope;
+import io.opentracing.mock.MockSpan;
+import io.opentracing.mock.MockTracer;
+import io.opentracing.tag.Tags;
+import io.opentracing.util.GlobalTracerTestUtil;
+import java.util.Optional;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.connection.RedisConnectionFactory;
+import org.springframework.data.redis.connection.jedis.JedisConnectionFactory;
+import org.springframework.data.redis.core.RedisTemplate;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * @author Daniel del Castillo
+ */
+@SpringBootTest(classes = {IntegrationTest.IntegrationTestConfiguration.class})
+@RunWith(SpringJUnit4ClassRunner.class)
+public class IntegrationTest {
+
+  @Configuration
+  @EnableAutoConfiguration
+  public static class IntegrationTestConfiguration {
+
+    public @Bean MockTracer mockTracer() {
+      GlobalTracerTestUtil.resetGlobalTracer();
+      return new MockTracer();
+    }
+
+    public @Bean RedisConnectionFactory redisConnectionFactory() {
+      JedisConnectionFactory jedisConnectionFactory = new JedisConnectionFactory();
+      jedisConnectionFactory.setHostName(redis.getHost());
+      jedisConnectionFactory.setPort(redis.getBindPort());
+      return jedisConnectionFactory;
+    }
+
+  }
+
+  private static RedisServer redis;
+
+  @Autowired
+  MockTracer tracer;
+
+  @Autowired
+  RedisTemplate redisTemplate;
+
+  @BeforeClass
+  public static void setup() throws Exception {
+    redis = RedisServer.newRedisServer();
+    redis.start();
+  }
+
+  @AfterClass
+  public static void teardown() {
+    redis.stop();
+  }
+
+  @Before
+  public void init() {
+    tracer.reset();
+  }
+
+  @Test
+  public void commandCreatesNewSpan() {
+    // TODO this fails on SB 1.5
+    redisTemplate.opsForValue().set(100L, "Some value here");
+    assertEquals(1, tracer.finishedSpans().size());
+    // TODO it's null here, but in SB 2.x it works
+    // assertEquals("SET", tracer.finishedSpans().get(0).tags().get("command"));
```

Comment:
@malafeev why the command is not set when using `opentracing-redis-spring-data`? I get this regression when compared to SB 2

Reply:
`command` is an `operationName` now, do we need to duplicate tag?

Reply:
yes